platform/lang-impl/src/com/intellij/ide/GeneratedFileEditingNotificationProvider.java

```diff
@@ -16,33 +15,25 @@
 import javax.swing.*;
 import java.util.function.Function;
 
+import static com.google.common.base.MoreObjects.firstNonNull;
+
 final class GeneratedFileEditingNotificationProvider implements EditorNotificationProvider, DumbAware {
   @Override
   public @Nullable Function<? super @NotNull FileEditor, ? extends @Nullable JComponent> collectNotificationData(@NotNull Project project,
                                                                                                                  @NotNull VirtualFile file) {
-    if (!GeneratedSourceFileChangeTracker.getInstance(project).isEditedGeneratedFile(file)) return null;
```

Comment:
@trushev I see @wgasiorek forgot to mention this when he published this PR, but the `GeneratedSourceFileChangeTracker` is now dead code. I haven't removed it as a part of this PR, but it probably should be removed. Thoughts?

Reply:
You are right. The dead code shall be removed. Let's do it as a separate patch just in case to be able to easily revert it